packages/ra-core/src/controller/button/useDeleteWithConfirmController.tsx

```diff
@@ -79,7 +79,42 @@ const useDeleteWithConfirmController = <RecordType extends RaRecord = any>(
     const notify = useNotify();
     const unselect = useUnselect(resource);
     const redirect = useRedirect();
-    const [deleteOne, { isLoading }] = useDelete<RecordType>();
+    const [deleteOne, { isPending }] = useDelete<RecordType>(
+        resource,
+        { id: record?.id },
```

Comment:
That was because it didn't matter as we specify them at call time, thus overriding them. I did that just to make TS happy as I needed to pass the options only